services/ui-src/src/components/menus/Sidebar.tsx

```diff
@@ -1,190 +1,251 @@
+      },
+      {
+        name: "III: Encounter Data Report",
+        path: "/encounter-data-report",
+      },
+      {
+        name: "X: Program Integrity",
+        path: "/program-integrity",
+      },
+    ],
   },
+
   {
     name: "C: Program-Level Indicators",
-    path: "program-level-indicators",
-    hasSubItems: true,
+    path: "/program-level-indicators",
+    children: null,
   },
   {
     name: "D: Plan-Level Indicators",
-    path: "plan-level-indicators",
-    hasSubItems: true,
+    path: "/plan-level-indicators",
+    children: null,
   },
-  { name: "E: BSS Entity Indicators", path: "bss-entity-indicators" },
-  { name: "Review & Submit", path: "review-and-submit" },
+  { name: "E: BSS Entity Indicators", path: "/bss-entity-indicators" },
+  { name: "Review & Submit", path: "/review-and-submit" },
 ];
 
 export const Sidebar = () => {
-  const [openMenuItemList, setOpenMenuItems] = useState([""]);
-  const toggleVisibility = (parentName: string) => {
-    if (openMenuItemList.includes(parentName)) {
-      setOpenMenuItems(openMenuItemList.filter((name) => name !== parentName));
-    } else {
-      setOpenMenuItems((state) => [...state, parentName]);
-    }
-  };
+  const { pathname } = useLocation();
+  const isMcparReport = pathname.includes("/mcpar");
+  const [isOpen, setIsOpen] = useState(true);
   return (
-    <Box sx={sx.root}>
-      <Flex sx={sx.sideNavTopFlex}>
-        <Text sx={sx.sideNavHeader}>MCPAR Report Submission Form</Text>
-        <CloseButton />
-      </Flex>
-      {LinkItems.map((link) => (
-        <NavItem
-          key={link.name}
-          path={link.path}
-          hasSubItems={!!link.hasSubItems}
-          name={link.name}
-          hidden={!!link.parent && !openMenuItemList.includes(link.parent)}
-          onClick={() => {
-            toggleVisibility(link.name);
-          }}
-        />
-      ))}
-    </Box>
+    <>
+      {isMcparReport && (
+        <Box sx={sx.root} className={isOpen ? "open" : "closed"}>
+          <Box
+            as="button"
+            sx={sx.closeButton}
+            onClick={() => setIsOpen(!isOpen)}
+          >
+            <ArrowIcon title="closeNavBarButton" direction="left" />
+          </Box>
+          <Box sx={sx.topBox}>
+            <Heading sx={sx.title}>MCPAR Report Submission Form</Heading>
+          </Box>
+          {LinkItems.map((section) => (
+            <NavSection
+              key={section.name}
+              section={section}
+              level={1}
+              basePath="/mcpar"
+            />
+          ))}
+        </Box>
+      )}
+    </>
   );
 };
 
-const CloseButton = () => {
+interface NavSectionProps {
+  key: string;
+  section: LinkItemProps;
+  level: number;
+  basePath: string;
+}
+
+const NavSection = ({ section, level, basePath }: NavSectionProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const { name, path, children } = section;
+  const itemPath = `${basePath}${path}`;
```

Comment:
This nifty little component will do the recursion for us. Each instance has its own open/closed state to control the `Collapse` that houses its children and accepts a base path from its parent that it can add its own path to.